pkg/runtime/reconciler.go

```diff
@@ -325,6 +325,17 @@ func (r *resourceReconciler) cleanup(
 		return err
 	}
 	if err = rm.Delete(ctx, observed); err != nil {
+		var requeueNeededAfter *requeue.RequeueNeededAfter
+		var requeueNeeded *requeue.RequeueNeeded
+		if errors.As(err, &requeueNeededAfter) ||
+			errors.As(err, &requeueNeeded) {
```

Comment:
I'm not in favor of using `errors.As` here, first because it's using the `reflect` library and second because it's a code that can panic if misused https://golang.org/src/errors/wrap.go?s=2494:2537#L67

`requeueNeededAfter` and `requeueNeeded` are nil pointers which will cause `errors.As` to panic

Reply:
You can use Go type assertions/switches for this use case. Something similar to:

```go
_, ok := err.(*requeue.RequeueNeededAfter)
if ok {
    // ...
}
```

Reply:
Updated code per the comment.